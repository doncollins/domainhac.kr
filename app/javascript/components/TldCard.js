import React, { forwardRef }  from 'react'
import DomainItem from './DomainItem'
import Loading from './Loading'

export default forwardRef(({ tld, domains }, ref) => {
    return (
        <div ref={ref} className="mb-8 w-full">
            <div className="text-4xl font-bold text-blue-500 mb-4 text-center">
                .{tld.label}
            </div>

            <div className="flex w-full justify-center">
                {domains && (
                    <div class="flex-grow max-w-xs sm:max-w-md shadow">
                        {domains.map(domain => (
                            <DomainItem key={domain.id} domain={domain} />
                        ))}
                    </div>
                ) || <Loading />}
            </div>
        </div>
    )
})