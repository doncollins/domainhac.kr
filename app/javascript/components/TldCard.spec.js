import React from 'react'
import renderer from 'react-test-renderer';
import mockNow from '../test/mockNow';
import TldCard from './TldCard';

beforeEach(mockNow)

it('renders correctly', () => {
    const tld = {
        id: 1,
        label: 'oo',
    }

    const domains = [
        {
            id: 1,
            label: 'f',
            tld: tld,
            availability: {
                id: 1,
                available: true,
                updated_at: new Date()
            },
        },        {
            id: 2,
            label: 'b',
            tld: tld,
            availability: {
                id: 2,
                available: false,
                updated_at: new Date()
            },
        }
    ]

    const component = renderer.create(
        <TldCard tld={tld} domains={domains} />
    )

    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
})