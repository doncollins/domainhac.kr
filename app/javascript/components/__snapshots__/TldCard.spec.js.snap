// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mb-8 w-full"
>
  <div
    className="text-4xl font-bold text-blue-500 mb-4 text-center"
  >
    .
    oo
  </div>
  <div
    className="flex w-full justify-center"
  >
    <div
      class="flex-grow max-w-xs sm:max-w-md shadow"
    >
      <div
        className="px-2 py-1 w-full flex items-center justify-between bg-gray-100 even:bg-white"
      >
        <i
          className="fa fa-check text-blue-500"
        />
        <div
          className="flex-grow ml-8 sm:ml-2 flex items-center justify-start"
        >
          <div
            className="sm:w-32 text-right"
          >
            f
          </div>
          <div
            className="text-blue-500"
          >
            .
          </div>
          <div>
            oo
          </div>
        </div>
        <div
          className="text-xs text-gray-500 flex items-center justify-end sm:w-1/2"
        >
          <span
            class="hidden sm:inline"
          >
            Updated 
          </span>
          <span>
            today at 6:14 PM
          </span>
        </div>
      </div>
      <div
        className="px-2 py-1 w-full flex items-center justify-between bg-gray-100 even:bg-white"
      >
        <i
          className="fa fa-check text-blue-500"
        />
        <div
          className="flex-grow ml-8 sm:ml-2 flex items-center justify-start"
        >
          <div
            className="sm:w-32 text-right"
          >
            b
          </div>
          <div
            className="text-blue-500"
          >
            .
          </div>
          <div>
            oo
          </div>
        </div>
        <div
          className="text-xs text-gray-500 flex items-center justify-end sm:w-1/2"
        >
          <span
            class="hidden sm:inline"
          >
            Updated 
          </span>
          <span>
            today at 6:14 PM
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
